source/serviceWorker.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-restricted-globals */
+
+// Cache necessary files when service worker is installed
+self.addEventListener('install', (e) => {
+  self.skipWaiting();
+  e.waitUntil(caches.open('ePomato').then((cache) => cache.addAll([
+    '/',
+    '/source/app.html',
+    '/source/done.html',
+    '/source/scss/main.scss',
+    '/source/js/scripts/registerSw.js',
+    '/source/js/scripts/backend.js',
+    '/source/js/scripts/app.js',
+    '/source/js/scripts/done.js',
+    '/source/js/components/StartContainer.js',
+    '/source/js/components/CurrentYear.js',
+    '/source/js/components/LogsButton.js',
+    '/source/js/components/StartButton.js',
+    '/source/js/components/UsernameInput.js',
+    '/source/js/components/WelcomeMessage.js',
+    '/source/js/classes/TaskList.js',
+    '/source/js/classes/PopUp.js',
+    '/source/img/green-tomato.ico',
+  ])));
+});
```

Comment:
So, this isn't going to work properly due to the fact that we are using parcel to compile and bundle our app. Thus, none of these files will actually exist in the build with these pathnames

Parcel has a plugin for this via: [parceljs-service-workers](https://parceljs.org/languages/javascript/#service-workers)

Just using the code snippet they provided seemed to work for me.

Reply:
Also, the fetch event listener will stay as is.